test/phoenix_live_view/router_test.exs

```diff
@@ -60,4 +60,14 @@ defmodule Phoenix.LiveView.RouterTest do
     assert Routes.foo_bar_nested_index_path(conn, :show) == "/router/foobarbaz/nested/show"
     assert Routes.custom_foo_bar_path(conn, :index) == "/router/foobarbaz/custom"
   end
+
+  test "user-defined metadata is available inside of metadata key" do
+    routes_with_metadata = Phoenix.LiveViewTest.Router.__routes__()
```

Comment:
This is private API IIRC. Please use `Phoenix.Router.route_info` to retrieve the route and its metadata for testing. :) If you have questions, see the commit that added `:metadata` to Phoenix!

Reply:
I didn't know it. I saw this looking for a way to retrieve all routes from the router.